tests/plugins/lti_consumer/test_cms_plugins.py

```diff
@@ -0,0 +1,182 @@
+
+
+@override_settings(
+    LTI_CONSUMER_SECRETS={
+        "marsha": {
+            "oauth_consumer_key": "InsecureOauthConsumerKey",
+            "shared_secret": "InsecureSharedSecret",
+        }
+    },
+    LTI_CONSUMER={
+        "display_name": "Marsha Video",
+        "pattern": "https://marsha.education.*",
+        "is_launch_url_regex": True,
+        "automatic_resizing": True,
+        "inline_ratio": 0.5625,
+    },
+)
+class LTIConsumerPluginTestCase(TestCase):
+    """Test suite for the lti consumer plugin."""
+
+    def test_cms_plugins_lti_consumer_context_and_html(self):
+        """
+        Instanciating this plugin with an instance should populate the context
+        and render in the template.
+        """
+        placeholder = Placeholder.objects.create(slot="test")
+
+        lti_consumer = LTIConsumerFactory()
+
+        model_instance = add_plugin(
+            placeholder,
+            LTIConsumerPlugin,
+            "en",
+            url=lti_consumer.url,
+            course=lti_consumer.course,
+        )
+        plugin_instance = model_instance.get_plugin_class_instance()
+        context = plugin_instance.render({}, model_instance, None)
+
+        # Check if "instance" is in context
+        self.assertIn("instance", context)
+
+        # Check if parameters, generated by the factory, are correctly set in "instance" of context
+        self.assertEqual(context["instance"].url, lti_consumer.url)
+
+        # Get generated html for lti consumer url
+        renderer = ContentRenderer(request=RequestFactory())
+        html = renderer.render_plugin(model_instance, {})
+
+        # Check rendered url is correct after save and sanitize
+        self.assertIn(lti_consumer.url, html)
+
+    def test_cms_plugins_lti_auth_parameters(self):
+        """
+        Verify that LTI authentication parameters are correctly build
+        """
+        lti_consumer = LTIConsumerFactory()
+        expected_auth_parameters = {
+            "lti_message_type": settings.LTI_CONSUMER.get("display_name"),
+            "lti_version": "LTI-1p0",
+            "resource_link_id": str(lti_consumer.id),
+            "context_id": lti_consumer.course.code,
+            "user_id": "richie",
+            "lis_person_contact_email_primary": "fun.dev@fun-mooc.fr",
+            "roles": "instructor",
+        }
+        self.assertDictEqual(lti_consumer.auth_parameters, expected_auth_parameters)
+
+    @mock.patch(
+        "richie.plugins.lti_consumer.models.oauth1.Client.sign",
+        return_value=(
+            "uri",
+            {
+                "Authorization": (
+                    'OAuth oauth_nonce="80966668944732164491378916897", '
+                    'oauth_timestamp="1378916897", '
+                    'oauth_version="1.0", oauth_signature_method="HMAC-SHA1", '
+                    'oauth_consumer_key="InsecureOauthConsumerKey", '
+                    'oauth_signature="frVp4JuvT1mVXlxktiAUjQ7%2F1cw%3D"'
+                )
```

Comment:
I don't think you should mock this but let it be computed and then check it because the entry parameters are known. Maybe take a look at this test suite: https://github.com/openfun/marsha/blob/ee6e325aa84ee1d65dc1c6b49019dcc80c9d6616/src/backend/marsha/core/tests/test_lti.py